core/primitives/src/errors.rs

```diff
@@ -304,17 +304,17 @@ pub enum ActionErrorKind {
     AddKeyAlreadyExists { account_id: AccountId, public_key: PublicKey },
     /// Account is staking and can not be deleted
     DeleteAccountStaking { account_id: AccountId },
-    /// Foreign sender (sender=!receiver) can delete an account only if a target account hasn't enough tokens to pay rent
-    DeleteAccountHasRent {
+    /// Foreign sender (sender=!receiver) can delete an account only if a target account hasn't enough tokens.
```

Comment:
This should never happen, correct? We wanted to avoid having someone's account being deletable by the third-party.

Reply:
You can create an account with 0 balance. We can't actually prevent this given each of our actions are executed independently at the moment.

Reply:
Then we should run `check_storage_cost ` on `actor_id` at the end of `apply_action_receipt`

Reply:
`check_storage_cost` is checked and enforced after `apply_action` call in `apply_action_receipt`  right before saving account back to storage. Unless we have a bug - there should be no changes to account post that.